src/datastore/src/main/java/org/locationtech/geogig/geotools/data/GeogigFeatureSource.java

```diff
@@ -247,6 +257,51 @@ protected int getCountInternal(Query query) throws IOException {
     protected FeatureReader<SimpleFeatureType, SimpleFeature> getReaderInternal(final Query query)
             throws IOException {
 
+        FeatureReader<SimpleFeatureType, SimpleFeature> featureReader;
+        featureReader = getNativeReader(query, true);
+
+        // do retyping to satisfy the Query's requested set of properties in the requested order
```

Comment:
unfortunately the geotools API requires the returned schema to be in the order requested by the Query property names, so that's what the retyping do.